File: techila/examples/python/Tutorial/1_distribution/run_distribution.py
# Copyright 2012-2013 Techila Technologies Ltd.

# This function contains the Local Control Code, which will create the
# computational Project.
#
# Usage:
# execfile('run_distribution.py')
# result = run_distribution(jobcount)
#
# jobcount: number of Jobs in the Project
#
# Example:
# result = run_distribution(5)
def run_distribution(jobcount):

    # Import the techila package
    import techila

    # Create the computational Project with the peach function.
    result = techila.peach(funcname = 'distribution_dist',   # Function that will be called on Workers
                           files = 'distribution_dist.py',   # Python-file that will be sourced on Workers
                           jobs = jobcount,                  # Number of Jobs in the Project
                           )

    # Display results after the Project has been completed. Each element
    # will correspond to a result from a different Job.
    print(result)
    return(result)

File: techila/examples/R/Tutorial/1_distribution/run_distribution.r
# Copyright 2010-2013 Techila Technologies Ltd.

# This function contains the Local Control Code, which will create the
# computational Project.
#
# Usage:
# source("run_distribution.r")
# result <- run_distribution(jobs)
# jobs: the number of Jobs in the Project
#
# Example:
# result <- run_distribution(5)
run_distribution <- function(jobs) {

  # Load the techila library
  library(techila)

  # Create the computational Project with the peach function.
  result <- peach(funcname = "distribution_dist",    # Function that will be called on Workers
                  files = list("distribution_dist.r"), # R-file that will be sourced on Workers
                  peachvector = 1:jobs,                 # Number of Jobs in the Project
                  sdkroot = "../../../..")              # Location of the techila_settings.ini file

  # Display results after the Project has been completed. Each element
  # will correspond to a result from a different Job.
  print(as.numeric(result))
  result
}
